encodings/fastlanes/src/bitpacking/compute/filter.rs

```diff
@@ -120,11 +123,15 @@ fn filter_indices<T: NativePType + BitPacking + ArrowNativeType>(
             }
         } else if indices_within_chunk.len() > UNPACK_CHUNK_THRESHOLD {
             // Unpack into a temporary chunk and then copy the values.
-            unsafe { BitPacking::unchecked_unpack(bit_width, packed, &mut unpacked) }
+            unsafe {
+                let dst: &mut [MaybeUninit<T>] = &mut unpacked;
```

Comment:
nit - I don't think you need this line, you can just transmute `&mut unpacked`

Reply:
This line is to avoid a lengthy transmute type hint. Essentially clippy complains if you transmute without type annotations and this is more readable imho